src/seq_mv/csr_matop_device.c

```diff
@@ -2282,15 +2440,88 @@ hypre_CSRMatrixTriLowerUpperSolveCusparse(char             uplo,
                           structural_zero, structural_zero);
             hypre_error_w_msg(1, msg);
          }
+#endif
       }
 
+#if CUSPARSE_VERSION >= CUSPARSE_SPSV_VERSION
+      HYPRE_CUSPARSE_CALL( cusparseSpSV_solve(handle, CUSPARSE_OPERATION_NON_TRANSPOSE,
+                                              &alpha, matA, vecF, vecU, data_type,
+                                              CUSPARSE_SPSV_ALG_DEFAULT,
+                                              hypre_CsrsvDataInfoU(csrsv_data)) );
+#else
       HYPRE_CUSPARSE_CALL( hypre_cusparse_csrsv2_solve(handle, CUSPARSE_OPERATION_NON_TRANSPOSE,
-                                                       nrow, nnzA, &alpha, descr, A_sa, A_i, A_sj,
+                                                       nrow, nnzA, &alpha, descr, A_ma, A_i, A_sj,
                                                        hypre_CsrsvDataInfoU(csrsv_data), f_data, u_data,
                                                        CUSPARSE_SOLVE_POLICY_USE_LEVEL,
                                                        hypre_CsrsvDataBuffer(csrsv_data)) );
+#endif
    }
 
+#if CUSPARSE_VERSION >= CUSPARSE_SPSV_VERSION
+   HYPRE_CUSPARSE_CALL( cusparseDestroySpMat(matA) );
+   HYPRE_CUSPARSE_CALL( cusparseDestroyDnVec(vecF) );
+   HYPRE_CUSPARSE_CALL( cusparseDestroyDnVec(vecU) );
+#endif
+
+   return hypre_error_flag;
+}
+
+
+HYPRE_Int
+hypre_CSRMatrixILU0(hypre_CSRMatrix *A)
```

Comment:
I think this is a great idea, but perhaps `seq_mv` is not the best place for it. I will introduce a `seq_ls` folder in another branch (currently WIP), and this new folder will probably be more suitable. I'm sharing this now so we don't forget in the future